rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -253,28 +350,66 @@ class InterpolatedValueCompilerImpl extends RegexParsers with InterpolatedValueC
 
-  def all: Parser[List[Token]] = (rep1(interpol | plainString) | emptyVar)
+  def all       : Parser[List[Token]] = (rep1(interpol   | plainString ) | emptyVar)
+  def noNodeProp: Parser[List[Token]] = (rep1(rudderProp | oldString   ) | emptyVar)
+
+  def space = """\s*""".r
 
   //empty string is a special case that must be look appart from plain string.
   //also parse full blank string, because no need to look for more case for them.
-  def emptyVar: Parser[List[CharSeq]] = """(?iums)(\s)*""".r ^^ { x => List(CharSeq(x)) }
+  def emptyVar    : Parser[List[CharSeq]] = """(?iums)(\s)*""".r ^^ { x => List(CharSeq(x)) }
 
-  def plainString: Parser[CharSeq] = ("""(?iums)((?!\Q${rudder.\E).)+""").r  ^^ { CharSeq(_) }
+  // plain string must not match our identifier, ${rudder.* and ${node.properties.*}
+  // here we defined a function to build them, with the possibility to
+  def plainString : Parser[CharSeq] = ("""(?iums)((?!(\Q${\E\s*rudder\s*\.|\Q${\E\s*node\s*\.\s*properties)).)+""").r  ^^ { CharSeq(_) }
+
+  // plain string old must not match our identifier, ${rudder.*
+  def oldString   : Parser[CharSeq] = ("""(?iums)((?!(\Q${rudder.\E)).)+""").r  ^^ { CharSeq(_) }
 
   //identifier for step in the path or param names
-  def propId: Parser[String] = """[\-_a-zA-Z0-9]+""".r
+  def propId      : Parser[String] = """[\-_a-zA-Z0-9]+""".r
+
+  // all interpolation
+  def interpol    : Parser[Interpolation] = rudderProp | nodeProp
 
   //an interpolated variable looks like: ${rudder.XXX}, or ${RuDder.xXx}
-  def interpol: Parser[Interpolation] = "${" ~> id("rudder") ~> "." ~> (nodeProp | parameter ) <~ "}"
+  def rudderProp  : Parser[Interpolation] = "${" ~> space ~> id("rudder") ~> space ~> id(".") ~> space ~>(nodeAccess | parameter ) <~ space <~ "}"
 
   //a node path looks like: ${rudder.node.HERE.PATH}
-  def nodeProp: Parser[Interpolation] = {
-    id("node") ~> "." ~> repsep(propId, ".") ^^ { seq => NodeAccessor(seq) }
-  }
+  def nodeAccess  : Parser[Interpolation] = { id("node") ~> space ~> id(".") ~> space ~> repsep(propId, space ~> id(".") ~> space) ^^ { seq => NodeAccessor(seq) } }
 
   //a parameter looks like: ${rudder.PARAM_NAME}
-  def parameter: Parser[Interpolation] = {
-    id("param") ~> "." ~> propId ^^ { p => Param(p) }
+  def parameter   : Parser[Interpolation] = { id("param") ~> space ~> id(".") ~> space ~> propId ^^ { p => Param(p) } }
+
+  //an interpolated variable looks like: ${rudder.XXX}, or ${RuDder.xXx}
+  def nodeProp    : Parser[Interpolation] = "${" ~> space ~> id("node") ~> space ~> id(".") ~> space ~> id("properties") ~> rep1( space ~> "[" ~> space ~>
+                                            propId <~ space <~ "]" ) ~ opt(propOption) <~ space <~ "}" ^^ { case ~(path, opt) => Property(path, opt) }
+
+  //here, the number of " must be strictly decreasing - ie. triple quote before
+  def propOption  : Parser[PropertyOption] = space ~> "|" ~> space ~>  ( onNodeOpt |  "default" ~> space ~> "=" ~> space ~>
```

Comment:
Lexing is such a mess that I can't really predict what it will look like. I prefer to keep it focused for now.